src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,82 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final String RESULT = "result";
+    public static final String SEPARATOR = ",";
+
+    private static class ReportData {
+        private int supply;
+        private int buy;
+
+        public int getSupply() {
+            return supply;
+        }
+
+        public void setSupply(int amount) {
+            this.supply += amount;
+        }
+
+        public int getBuy() {
+            return buy;
+        }
+
+        public void setBuy(int amount) {
+            this.buy += amount;
+        }
+    }
+
+    private ReportData readAndCalculateData(String fromFileName) {
+        ReportData data = new ReportData();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            String category;
+            int amount;
 
+            while ((line = br.readLine()) != null) {
+                String[] lineArr = line.split(SEPARATOR);
+                category = lineArr[0];
+                amount = Integer.parseInt(lineArr[1]);
+
+                if (category.equals(SUPPLY)) {
+                    data.setSupply(amount);
+                } else if (category.equals(BUY)) {
+                    data.setBuy(amount);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

This violates checklist item #11: 'Don't ignore exceptions'. The requirement states to re-throw a `RuntimeException` with a descriptive message and the original exception. The current implementation is missing the descriptive message, which should be something like `"Can't read data from the file " + fromFileName`.